src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -1319,15 +1319,15 @@ TEST_F(ClusterTaskManagerTest, BacklogReportTest) {
   ASSERT_EQ(pool_.workers.size(), 0);
   ASSERT_EQ(node_info_calls_, 0);
 
-  {  // No tasks can run because the worker pool is empty.
+  {  // 1 task has resources allocated, while remaining 9 is stuck.
```

Comment:
this is also similar,
before:
- if a task has resources allocated, but no worker, it will be reported as demand. 
- however, this will lead to scaling up in autoscaler as autoscaler thought there's no enough resources, but there's actually just delay in starting the worker. 

now: 
- once a task has resource allocated, it will not show up as demand, which makes a node's resource usage consistent with the load demand.

Reply:
so this test was originally written in a way one of tasks are assigned to worker already?

Reply:
originally the code and the test are both double-counting a dispatching task with resources allocated as pending, which is the root-cause of the inconsistency. 

in the original code, only when worker is ready, a dispatch task will then be removed from the demand, but it should be removed from the demand when resources could be allocated. (And it will be added back as demand when worker popping fails)